airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/destination/StandardNameTransformer.java

```diff
@@ -69,4 +74,21 @@ public static JsonNode formatJsonPath(final JsonNode root) {
     }
   }
 
+  @Override
+  public String getStageName(final String namespace, final String streamName) {
+    return applyDefaultCase(String.join("_", convertStreamName(namespace), convertStreamName(streamName)));
+  }
+
+  @Override
+  public String getStagingPath(final String connectionId, final String namespace, final String streamName, final DateTime writeDatetime) {
+    // see https://docs.snowflake.com/en/user-guide/data-load-considerations-stage.html
+    return applyDefaultCase(String.format("%s/%s/%s/%02d/%02d/%02d/",
+        connectionId,
+        getStageName(namespace, streamName),
+        writeDatetime.year().get(),
+        writeDatetime.monthOfYear().get(),
+        writeDatetime.dayOfMonth().get(),
+        writeDatetime.hourOfDay().get()));
+  }
```

Comment:
This is very specific to snowflake.

I think it's better to move the two stage methods to `StagingOperations`.